R/validation_asserts.R

```diff
@@ -312,3 +312,15 @@ assert_file_extension <- function(file, types) {
   }
   invisible(TRUE)
 }
+
+assert_single_level <- function(data) {
+  levels <- lapply(data$area_id, function(area_id) {
+    gsub("\\w{3}_(\\d+)_\\d+.*", "\\1", area_id)
```

Comment:
Do we use this regex elsewhere for checking?  I would not require the third block to be digits -- that will fail for South Africa; I think also maybe fails on Nigeria.

Reply:
@r-ash revised check to use `area_level` from areas dataset instead of string parsing